crates/bevy_pbr/src/fog.rs

```diff
@@ -0,0 +1,171 @@
+    pub mode: FogMode,
+}
+
+/// Allows switching between different the [`Fog`] “modes”, and configuring their parameters.
+#[derive(Debug, Clone, Default, Reflect)]
+pub enum FogMode {
+    /// Fog effect is disabled (the default)
+    #[default]
+    Off,
+
+    /// A linear fog effect that grows in intensity between `start` and `end` distances.
+    ///
+    /// This mode is simpler to control than other modes, however it can produce results that look “artificial”, depending on the scene.
+    ///
+    /// ## Formula
+    ///
+    /// The fog intensity for a given point in the scene is determined by the following formula:
+    ///
+    /// ```text
+    /// let fog_intensity = 1.0 - ((end - distance) / (end - start)).clamp(0.0, 1.0);
+    /// ```
+    ///
+    /// <svg width="370" height="212" viewBox="0 0 370 212" fill="none">
+    /// <title>Plot showing how the linear fog mode behaves for start and end values of 0.8 and 2.2, respectively.</title>
+    /// <path d="M331 151H42V49" stroke="currentColor" stroke-width="2"/>
+    /// <text font-family="sans-serif" fill="currentColor" style="white-space: pre" font-family="Inter" font-size="12" letter-spacing="0em"><tspan x="136" y="173.864">1</tspan></text>
+    /// <text font-family="sans-serif" fill="currentColor" style="white-space: pre" font-family="Inter" font-size="12" letter-spacing="0em"><tspan x="30" y="53.8636">1</tspan></text>
+    /// <text font-family="sans-serif" fill="currentColor" style="white-space: pre" font-family="Inter" font-size="12" letter-spacing="0em"><tspan x="42" y="173.864">0</tspan></text>
+    /// <text font-family="sans-serif" fill="currentColor" style="white-space: pre" font-family="Inter" font-size="12" letter-spacing="0em"><tspan x="232" y="173.864">2</tspan></text>
+    /// <text font-family="sans-serif" fill="currentColor" style="white-space: pre" font-family="Inter" font-size="12" letter-spacing="0em"><tspan x="332" y="173.864">3</tspan></text>
+    /// <text font-family="sans-serif" fill="currentColor" style="white-space: pre" font-family="Inter" font-size="12" letter-spacing="0em"><tspan x="161" y="190.864">distance</tspan></text>
+    /// <text font-family="sans-serif" transform="translate(10 132) rotate(-90)" fill="currentColor" style="white-space: pre" font-family="Inter" font-size="12" letter-spacing="0em"><tspan x="0" y="11.8636">fog intensity</tspan></text>
+    /// <path d="M43 150H117.227L263 48H331" stroke="#FF00E5"/>
+    /// <path d="M118 151V49" stroke="#FF00E5" stroke-dasharray="1 4"/>
+    /// <path d="M263 151V49" stroke="#FF00E5" stroke-dasharray="1 4"/>
+    /// <text font-family="sans-serif" fill="#FF00E5" style="white-space: pre" font-family="Inter" font-size="10" letter-spacing="0em"><tspan x="121" y="58.6364">start</tspan></text>
+    /// <text font-family="sans-serif" fill="#FF00E5" style="white-space: pre" font-family="Inter" font-size="10" letter-spacing="0em"><tspan x="267" y="58.6364">end</tspan></text>
+    /// </svg>
```

Comment:
TIL you can embed SVG graphs inside of rustdoc!

Reply:
I saw the `Color` constant docs that use styled `<span>`s and decided to give it a try, was also pleasantly surprised